examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py

```diff
@@ -113,6 +115,43 @@ def get_migration_state(self) -> AirflowMigrationState:
                 dag_dict[dag_id] = DagMigrationState.from_dict(migration_dict)
         return AirflowMigrationState(dags=dag_dict)
 
+    def get_task_instance_batch(
+        self, dag_id: str, task_ids: Sequence[str], run_id: str, states: Sequence[str]
+    ) -> List["TaskInstance"]:
+        """Get all task instances for a given dag_id, task_ids, and run_id."""
+        task_instances = []
+        task_id_chunks = [
+            task_ids[i : i + BATCH_GET_TASK_INSTANCES_LIMIT]
+            for i in range(0, len(task_ids), BATCH_GET_TASK_INSTANCES_LIMIT)
+        ]
+        for task_id_chunk in task_id_chunks:
+            response = self.auth_backend.get_session().post(
+                f"{self.get_api_url()}/dags/~/dagRuns/~/taskInstances/list",
```

Comment:
what's going on with the tildes here? Does this magically do string substitution somehow?

Reply:
https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/get_dag_runs_batch